tools/deploy/planner.go

```diff
@@ -135,6 +136,8 @@ func (p deploymentPlanner) generateStepFromTargetType(
 			delete(instances, shardSetID)
 			continue
 		}
+
+		done := false
```

Comment:
nit: use default initialization, ie `var done bool`

Reply:
Yeah I tend to be more explicit in cases like this, feels more clear to the readers.

Reply:
nit: it feels cleaner to create a generateSortedStepFromTargetType that sorts the step generated from generateStepFromTargetType

Reply:
Hm that feels like a bit overkill, this is only sorting instances within the same step so whether they sorted or not doesn't affect correctness of the output, so might as well always sort them for more deterministic output.

Reply:
sounds good